### PR TITLE
#1284 Implement Version Number pattern

### PR DESCRIPTION
**Pull request by manannikov**
Implemented Version Number pattern described in issue #1284

### CODE REVIEW

File: version-number/src/test/java/com/iluwatar/versionnumber/BookRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    try {
+      bookRepository.update(bobBook);
+    } catch (VersionMismatchException e) {
+      assertEquals(0, bobBook.getVersion());
+      assertEquals(1, bookRepository.get(bookId).getVersion());
+      assertEquals(aliceBook.getVersion(), bookRepository.get(bookId).getVersion());
+      assertEquals("", bobBook.getTitle());
+      assertNotEquals(aliceBook.getAuthor(), bobBook.getAuthor());
+    }
+  }
```
Comment (iluwatar) [inline version-number/src/test/java/com/iluwatar/versionnumber/BookRepositoryTest.java:71]: I think this single test could be divided into multiple smaller ones. Each test should have a meaningful name that directly explains what is being tested e.g. `shouldThrowBookDuplicateException`.

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: version-number/src/main/java/com/iluwatar/versionnumber/App.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  /**
+   * Program entry point.
+   *
+   * @param args command line args
+   */
+  public static void main(String[] args) throws
+      BookDuplicateException,
+      BookNotFoundException,
+      VersionMismatchException {
+    long bookId = 1;
```
[the remark this answers is not in the document]
Comment (manannikov) [reply version-number/src/main/java/com/iluwatar/versionnumber/App.java]: done

File: version-number/src/test/java/com/iluwatar/versionnumber/BookRepositoryTest.java (hunk 1 above)
Comment (manannikov) [reply version-number/src/test/java/com/iluwatar/versionnumber/BookRepositoryTest.java]: done

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Huge thanks for the new pattern! 💪🏻 

@all-contributors please add @manannikov for code

**Pull request merged**